philology/src/main/java/com/jcminarro/philology/ResourcesUtil.kt

```diff
@@ -12,36 +12,39 @@ internal class ResourcesUtil(private val baseResources: Resources) {
 
     @Throws(Resources.NotFoundException::class)
     fun getText(id: Int): CharSequence {
-        return repository.getText(Resource.Text(baseResources.getResourceEntryName(id)))
+        return repository.getText(baseResources.getResourceEntryName(id))
             ?: baseResources.getText(id)
     }
 
     @Throws(Resources.NotFoundException::class)
     fun getString(id: Int): String = getText(id).toString()
 
     @Throws(Resources.NotFoundException::class)
-    fun getQuantityText(id: Int, quantity: Int): CharSequence = repository.getText(
-        Resource.Plural(
-            baseResources.getResourceEntryName(id),
-            quantity.toPluralKeyword(baseResources)
-        )
+    fun getQuantityText(id: Int, quantity: Int): CharSequence = repository.getPlural(
+        baseResources.getResourceEntryName(id),
+        quantity.toPluralKeyword(baseResources)
     ) ?: baseResources.getQuantityText(id, quantity)
 
     @Throws(Resources.NotFoundException::class)
     fun getQuantityString(id: Int, quantity: Int): String = getQuantityText(id, quantity).toString()
 
     @Throws(Resources.NotFoundException::class)
-    fun getQuantityString(id: Int, quantity: Int, vararg formatArgs: Any?): String =
-        String.format(getQuantityString(id, quantity), *formatArgs)
-}
+    fun getQuantityString(id: Int, quantity: Int, vararg formatArgs: Any?): String {
+        return String.format(getQuantityString(id, quantity), *formatArgs)
+    }
 
-interface PhilologyRepository {
-    fun getText(resource: Resource): CharSequence?
+    fun getStringArray(id: Int) = getTextArray(id).map { it.toString() }.toTypedArray()
+
+    fun getTextArray(id: Int): Array<CharSequence> {
+        return repository.getTextArray(baseResources.getResourceEntryName(id))
+            ?: baseResources.getTextArray(id)
+    }
 }
 
-sealed class Resource(open val key: String) {
-    data class Text(override val key: String) : Resource(key)
-    data class Plural(override val key: String, val quantityKeyword: String) : Resource(key)
+interface PhilologyRepository {
```

Comment:
I think that change is great, the final developer need to know less types to create the repository.
Because the Public API of the `PhilologyRepository` has changed, could you update the examples on the `README.md` file?